libbeat/logp/logp.go

```diff
@@ -141,3 +156,57 @@ func getLogLevel(config *Logging) (Priority, error) {
 	}
 	return level, nil
 }
+
+// snapshotExpvars iterates through all the defined expvars, and for the top
+// level vars that are integers it snapshots the name and value in a separate
+// map.
+func snapshotExpvars(varsMap map[string]int64) {
+	expvar.Do(func(kv expvar.KeyValue) {
+		switch kv.Value.(type) {
+		case *expvar.Int:
```

Comment:
I think this PR covers the most important metrics for the time being so no action is required at the moment IMO. We can improve this when we add "metrics" within beats. We have some expvar metrics that are nested within a map that won't be printed by this. For example, in Metricbeat we have a `fetches.mysql-info.failures` metric that is nested two levels deep in maps.